pkg/controller/networkpolicy/validate.go

```diff
@@ -36,15 +69,40 @@ var (
 	// reservedTierNames stores the set of Tier names which cannot be deleted
 	// since they are created by Antrea.
 	reservedTierNames = sets.NewString("baseline", "application", "platform", "networkops", "securityops", "emergency")
+	// apv is an instance of antreaPolicyValidator to validate Antrea-native
+	// policy events.
+	apv = antreaPolicyValidator{}
+	// tv is an instance of tierValidator to validate Tier resource events.
+	tv = tierValidator{}
 )
 
+// registerAntreaPolicyValidator registers an Antrea-native policy validator to
+// the resource registry.
+func registerAntreaPolicyValidator(v *validatorRegistry, a validator) {
+	v.antreaPolicyValidators = append(v.antreaPolicyValidators, a)
+}
+
+// registerTierValidator registers a Tier validator to the resource registry.
+func registerTierValidator(v *validatorRegistry, t validator) {
+	v.tierValidators = append(v.tierValidators, t)
+}
+
 type NetworkPolicyValidator struct {
+	// reg maintains list of validators to be called during validation of
+	// Antrea-native policies and Tier resources.
+	reg                     *validatorRegistry
 	networkPolicyController *NetworkPolicyController
 }
 
 // NewNetworkPolicyValidator returns a new *NetworkPolicyValidator.
 func NewNetworkPolicyValidator(networkPolicyController *NetworkPolicyController) *NetworkPolicyValidator {
```

Comment:
I notice that there is no public API to register `validator` objects? Does that mean that the only way to add new ones is to edit this source file?